### ISSUE TITLE
Nested configurations

### ISSUE THREAD

**Issue by sidnarayanan** [fidelity/spock#6]
Allow for nested configurations (with the full suite of type-checking, etc). The syntax should look something like this:

```python

@spock_config 
ObjectConfig:
    param_1a: int 
    param_1b: float

@spock_config
RepeatedObjectConfig:
    param_2: float

@spock_config
ParentConfig:
    param_3: int
    object: ObjectConfig
    repeated_objects: List[RepeatedObjects]
```

The corresponding YAML will look like:
```yaml
param_3: 5
object:
    param_2: 3.14
repeated_objects:
    - param_1a: 2
      param_1b: 2.71
    - param_1a: 4
      param_1b: 1.51
```

The corresponding command line args (#5 ) will look like:
```
./executable.py --object.param_2 3.14 
```
I'm not sure about the syntax for lists of configs in the command line, but we could say those have to be defined in YAMLs

**Comment by mhanco** [comment]
Something like the `jq` filter syntax would work for yaml as well.
```
{
  "param_3": 5,

  "object": {"param_2": 3.14},
  "repeated_objects":[
    {
      "param_1a": 2, 
      "param_1b": 2.71
    },
    {
      "param_1a": 4,
      "param_1b": 1.51
    }
  ]
}
```
Command line
```
jq '.repeated_objects[0].param_1b' nested.json
2.71
```

**Comment by ncilfone** [comment]
Closed by #27
